### ISSUE TITLE
[models] Add model compression utils

### ISSUE THREAD

**Issue by fg-mindee** [mindee/doctr#5]
Add a `doctr.models.utils` module to compress existing models and improve their latency / memory load for inference purposes on CPU. Some interesting leads to investigate:
- [x] FP conversion (#10)
- [x] Quantization (#10)
- [ ] Pruning (cf. https://www.tensorflow.org/model_optimization/guide/pruning/comprehensive_guide)
- [x] TF Lite export (#10)
- [x] ONNX export (cf. https://github.com/onnx/keras-onnx & https://github.com/onnx/tensorflow-onnx)
- [x] Export to SaveModel (#246)

_Optional: TensorRT export (cf. https://developer.nvidia.com/blog/speeding-up-deep-learning-inference-using-tensorflow-onnx-and-tensorrt/)_

**Comment by fg-mindee** [comment]
A good lead for ONNX support would be to use https://github.com/onnx/tensorflow-onnx (might have to create a savemodel to use it but it's worth a look)

**Comment by felixdittrich92** [comment]
@frgfm i think we can remove the tensorrt point If we support onnx wdyt ?

**Comment by frgfm** [comment]
Yes sure! We'll need to take a look at pruning at some point

**Comment by felixdittrich92** [comment]
yeah pruning is fine but tensorrt is a bit to much (should do the user on his own side if we can provide onnx this should be not so tricky)

**Comment by felixdittrich92** [comment]
@frgfm @sebastianMindee @SiddhantBahuguna 

I think that's something we can close:

We have:

- Onnx export - both TF & PyTorch
- FP16 support - train (AMP) & inference - TF & PyTorch
- `torch.compile` support pyTorch

wdyt ?
